### PR TITLE
Development

### PR DESCRIPTION
**Pull request by apmccartney**
Do not pull. Pushed for code review purposes

### CODE REVIEW

File: implementation/record/Tail.hpp.in [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+template < typename tail >
+using MF = TailVerification< 1, tail >;
+
+template < typename tail >
+using MT = TailVerification< 2, tail >;
+
+/* Convenience class providing a type definition corresponding to the 
+ * composition of zero or more component verifiers 
+ */
+template< template< typename > class... Ts >
```
Comment (jlconlin) [inline implementation/record/Tail.hpp.in]: Why are these not in their own header/directory? It seems we are moving away from our structured hierarchy.

File: implementation/StructureDivision.hpp.in [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      ENDFtk::implementation::record::Integer<11>,
+      ENDFtk::implementation::record::Integer<11>,
+      ENDFtk::implementation::record::Integer<11> >;
+
+  using Tail = ENDFtk::implementation::record::Tail;
+
+  Base base;
+  Tail tail;
+
+  StructureDivision( double C1, double C2, long L1, long L2, long N1, long N2,
```
Comment (jlconlin) [inline implementation/StructureDivision.hpp.in]: Gotta have documentation here. StructureDivision isn't part of ENDF so it needs to have an explanation of what it is.

File: implementation/StructureDivision.hpp.in [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    throw e;
+  } catch ( int fieldNumber ){
+    throw std::exception();
+  }
+
+  bool isHead(){
+    return this->tail.section() && this->tail.file() && this->tail.material();
+  }
+  
+  bool isSendPermissive(){
```
Comment (jlconlin) [inline implementation/StructureDivision.hpp.in]: I'm not sure what permissive means here.

File: implementation/HeadRecord/src/asHead.hpp.in [hunk 4]
```diff
@@ -0,0 +1,63 @@
+friend HeadRecord&
```
Comment (jlconlin) [inline implementation/HeadRecord/src/asHead.hpp.in]: Why does this (and the following) need to be a friend?

File: implementation/record/Tail.hpp.in (hunk 1 above)
Comment (apmccartney) [inline implementation/record/Tail.hpp.in]: I believe you wrote this file; I just moved it into another namespace

File: implementation/HeadRecord/src/asHead.hpp.in (hunk 4 above)
Comment (apmccartney) [inline implementation/HeadRecord/src/asHead.hpp.in]: protected inheritance

**Pull request merged**